### PR TITLE
Directional commands

### LINKED ISSUE #2: Simple Direction Commands
# Offboard Directional Commands

## Problem

Need some simple offboard commands to maneuver the drone quickly and efficiently without the use of higher-level MAVSDK functions. Basic movement single-directional movement across all 3 axes, as well as some diagonal commands, is desired.

## Solution

Utilize the offboard commands provided by MAVSDK to directly control the drone. Learn the parameters required for each function and how each different offboard command alters the drone's orientation.

### LINKED ISSUE #3: Unit Tests for Offboard Commands
# Unit Tests for Offboard Commands

## Problem

Need simple unit tests to ensure the offboard movement controls the drone in the manner we intend/need.

## Solution

The unit tests should test each movement individually, with different files for each axis.

## Additional Information

The unit tests should call each function in a controlled environment, logging which functions fail.

### PR DESCRIPTION
**Pull request by DJDMVP**
Basic Directional commands for reference for object avoidance. Uses velocity to move a certain distance based on drone body. There are a couple movement commands including in each direction, in a diagonal, and rotating.

### CODE REVIEW

File: Commands/forward.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    Parameters
+    ----------
+    drone : System
+        MavSDK object for drone control
+    distance: float
+        distance in meters wanted to go forward
+    """
+
+    ms_speed: float = 20
+    move_time: float = distance / 20
```
Comment (cjhr95) [inline Commands/forward.py]: Change the denominator to ms_speed

File: Commands/left.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    Parameters
+    ----------
+    drone : System
+        MavSDK object for drone control
+    distance: float
+        distance in meters wanted to go left
+    """
+
+    ms_speed: float = 20
+    move_time: float = distance / 20
```
Comment (cjhr95) [inline Commands/left.py]: Change the denominator to ms_speed

File: Commands/movement.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+"""
+import asyncio
+import math
+from mavsdk import System
+
+from mavsdk import offboard
+
+
+async def directional_movement(
+    drone: System, rightdistance: float, forwarddistance: float, updistance: float
```
Comment (cjhr95) [inline Commands/movement.py]: Add some underscores to these parameters to adhere to snake_case

File: Commands/movement.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    Parameters
+    ----------
+    drone : System
+        MavSDK object for drone control
+    rightdistance: float
+        distance in meters wanted to go right
+    forwarddistance: float
+        distance in meters wanted to go forward
+    updistance: float
+        distance in meters wanted to go up
```
Comment (cjhr95) [inline Commands/movement.py]: Same here, change parameter names to match the function definition

File: Commands/test.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+async def run() -> None:
+    """ """
+
+    # create a drone object
+    drone: System = System()
+    await drone.connect(system_address=SIM_ADDR)
+
+    # initilize drone configurations
+    await drone.action.set_takeoff_altitude(12)
+    await drone.action.set_maximum_speed(25)
```
Comment (cjhr95) [inline Commands/test.py]: Have these parameters be set to constants for readability & easy edits

File: Commands/test.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    await asyncio.sleep(30)
+
+    await rotate(drone, 90)
+
+    await asyncio.sleep(30)
+
+    await forward(drone, 100)
+
+    await asyncio.sleep(30)
+    """
```
Comment (cjhr95) [inline Commands/test.py]: Remove the large comment block

**Review by cjhr95 — CHANGES_REQUESTED** [review]
Just some small stuff on variable naming

File: Commands/test.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+from mavsdk import System
+
+from backward import backward
+from forward import forward
+
+# from left import left
+# from right import right
+# from rotate import rotate
+# from upward import upward
+# from downward import downward
```
Comment (cjhr95) [inline Commands/test.py]: Remove the unused imports

File: Commands/test.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    """
+    tests different movement commands
+    """
+
+    # create a drone object
+    drone: System = System()
+    await drone.connect(system_address=SIM_ADDR)
+
+    # initilize drone configurations
+
```
Comment (cjhr95) [inline Commands/test.py]:
```suggestion
```

**Review by cjhr95 — CHANGES_REQUESTED** [review]
Some very small, itty-bitty formatting stuff

**Comment by RexBerry** [comment]
Forgot the emoticon in the commit message:
👁️👄👁️

**Review by cjhr95 — APPROVED** [review]
Yeah it's alright or whatever

**Pull request merged**